applications/build-modular-monolith-with-platformatic/services/media-service/plugin.js

```diff
@@ -0,0 +1,75 @@
+/// <reference path="./clients/persons/persons.d.ts" />
+
+'use strict'
+
+/** @param {import('fastify').FastifyInstance} app */
+module.exports = async function (app) {
+  function buildOnComposerResponseCallback (personsProps) {
+    return async function addPersonsToResponse (request, reply, body) {
+      let entities = await body.json()
+
+      const multipleEntities = Array.isArray(entities)
+      if (!multipleEntities) {
+        entities = [entities]
+      }
+
+      const personIds = []
+      for (const entity of entities) {
+        for (const { idProp } of personsProps) {
+          personIds.push(entity[idProp])
+        }
+      }
+
+      const persons = await app.persons.getPersons({ "where.id.in": personIds.join(',') })
+
+      const getPersonNameById = (id) => {
+        const person = persons.find(person => person.id === id)
+        return (person) ? person.name : null
+      }
+
+      for (let entity of entities) {
+        for (const { idProp, nameProp } of personsProps) {
+          entity[nameProp] = getPersonNameById(entity[idProp])
+        }
+      }
+
+      reply.send(multipleEntities ? entities : entities[0])
+    }
+  }
+
+  function booksOnRouteHook (routeOptions) {
+    // TODO: These schema changes apply to the `/books/{id}` route too
+    //       Bug: It seems like this is changing a shared schema by reference
+    const responseSchema = routeOptions.schema.response[200].items
+    if (responseSchema && !responseSchema.properties.authorName) {
+      responseSchema.properties.authorName = {
+        type: 'string'
+      }
+      responseSchema.required.push('authorName')
+    }
```

Comment:
This code is first run for the `/books/` route and then for the `/books/{id}` route. The `responseSchema.properties.authorName` check is needed here, as when this code is run the second time (for the `/books/{id}` route) it will end up modifying the response schema so that it's invalid (`authorName` will exist twice in the `required` array).

This suggests to me that the response schemas for the `/books/` and `/books/{id}` routes are actually a single object.

Reply:
It should be two different objects.

Reply:
I can confirm that the composer passes the same object in some cases. I will take a look. I can't tell right now where it should be handled: inside of the composer or directly in the hook.